packages/camera/camera/lib/src/camera_preview.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of camera;
+
+/// A widget showing a live camera preview.
+class CameraPreview extends StatelessWidget {
+  /// Creates a preview widget for the given camera controller.
+  const CameraPreview(this.controller);
+
+  /// The controller for the camera that the preview is shown for.
+  final CameraController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return controller.value.isInitialized
+        ? Texture(textureId: controller._cameraId)
+        : Container();
```

Comment:
Done ;)

I do wonder if the `CameraPreview` widget can actually become a cross platform widget since it relies on the `CameraController` which is a Android / iOS implementation of the camera_platform_interface. I think we can also make the `CameraController` cross platform but not sure if this will work for the web (looking at the  `google_maps_flutter` I also see separate controllers and widgets). What do you think @ditman?

Reply:
Everything that is exposed in the `camera/camera` must be completely cross-platform, and if it needs to do calls to the native side, it needs to use a platform interface call. I thought the controller was already doing that? Re-checking!

Reply:
The current implementation of the camera controller is using a method channel directly, but I think all the methods called there are already represented in the platform_interface?

Reply:
The current camera controller is using the method channel directly because I reverted the changes in the `camera/camera` plugin. Changes for the `camera/camera` plugin are now part of PR #3302, here all calls to the native side go through the platform_interface (except for the `startImageStream` and `stopImageStream`).